addons/purchase_stock/tests/test_replenish_wizard.py

```diff
@@ -273,6 +273,44 @@ def test_chose_supplier_4(self):
         self.assertEqual(last_po_id.partner_id, vendor1)
         self.assertEqual(last_po_id.order_line.price_unit, 60)
 
+    def test_chose_supplier_5(self):
+        """ Choose supplier based on discounted price
+        replenish 1
+
+        1)seq1 vendor 100 discount 10%
+        2)seq2 vendor 110 discount 20%
+        -> 2) should be chosen
+        """
+        self.supplierinfo.product_tmpl_id = self.product1.product_tmpl_id.id
```

Comment:
You shouldn't need this line since `seller_ids` in product_template is One2many (the link is already there from when cls.product1 was created and product.product extends product.template). It's not a big deal though, just for your future reference.